.github/workflows/sonarqube.yml

```diff
@@ -38,8 +42,10 @@ jobs:
             -Dsonar.projectKey=com.github.spotbugs:sonar-findbugs-plugin \
             -Dsonar.organization=spotbugs \
             -Dsonar.host.url=https://sonarcloud.io \
-            -Dsonar.login=$SONAR_LOGIN
+            -Dsonar.login=$SONAR_LOGIN \
+            ${PR_NUMBER:+ -Dsonar.pullrequest.key=$PR_NUMBER -Dsonar.pullrequest.branch=${{ github.event.pull_request.head.ref }} }
```

Comment:
`${{ github.event.pull_request.head.ref }}` is untrusted value,
We should avoid to use this untrusted value in `run:`.

For example, an attcker can create a branch that is `""}|echo${IFS}${SONAR_LOGIN}|base64;${IFS`.
This workflow output secrets.SONAR_LOGIN in GitHub Actions's log.

- PR example: https://github.com/azu/sonar-findbugs-pull-384/pull/10
- Log: https://github.com/azu/sonar-findbugs-pull-384/pull/10/checks?check_run_id=3457998931

<img width="1160" alt="image" src="https://user-images.githubusercontent.com/19714/131282220-80581568-b4a3-4ca4-a9c6-fbbc7d6a075c.png">

📝 GitHub Actions mask the same value to secrets, so I avoid it using `base64` command

[Keeping your GitHub Actions and workflows secure Part 2: Untrusted input | GitHub Security Lab](https://securitylab.github.com/research/github-actions-untrusted-input/) describe this vulnerbility.
Also, [actionlint](https://github.com/rhysd/actionlint) can found this vulerbility.

To fix this issue, get `github.event.pull_request.head.ref` value via env.

```diff
diff --git a/.github/workflows/sonarqube.yml b/.github/workflows/sonarqube.yml
index 6218e7c..896cd3d 100644
--- a/.github/workflows/sonarqube.yml
+++ b/.github/workflows/sonarqube.yml
@@ -43,9 +43,10 @@ jobs:
             -Dsonar.organization=spotbugs \
             -Dsonar.host.url=https://sonarcloud.io \
             -Dsonar.login=$SONAR_LOGIN \
-            ${PR_NUMBER:+ -Dsonar.pullrequest.key=$PR_NUMBER -Dsonar.pullrequest.branch=${{ github.event.pull_request.head.ref }} }
+            ${PR_NUMBER:+ -Dsonar.pullrequest.key=$PR_NUMBER -Dsonar.pullrequest.branch=${PR_BRANCH} }
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SONAR_LOGIN: ${{ secrets.SONAR_LOGIN }}
           PR_NUMBER: ${{ github.event.pull_request.number }}
+          PR_BRANCH: ${{ github.event.pull_request.head.ref }}
           CI: true
```

Reply:
Thanks @azu for your review!